mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -692,6 +697,20 @@ LogicalResult getTuningProblemStr(rock::RockGemmWrapperInterface gemmIF,
     oLayout[oLayoutMap["1o"]] = '1';
     oLayout[oLayoutMap["go"]] = 'G';
 
+    if (size > 5) {
+      for (unsigned i = 0; i < size - 5; i++) {
+        std::string key = std::to_string(i + 2);
+        fLayout[fLayoutMap[key]] = '0' + i + 2;
+        iLayout[iLayoutMap[key + "i"]] = '0' + i + 2;
+        oLayout[oLayoutMap[key + "o"]] = '0' + i + 2;
+      }
+    }
+
+    if (llvm::any_of(llvm::concat<const char>(fLayout, iLayout, oLayout),
+                     [](const char c) { return c == '#'; })) {
+      llvm::report_fatal_error("Tuning problem string not properly filled.");
```

Comment:
return failure();

Reply:
Okay, but it just means that the next program will barf.

Reply:
The point is that we don't raise asserts or otherwise crash the program on invalid user input to the extend it's reasonable to do so